# **ESTUDO DE CASO PARA MATÉRIA DE MINERAÇÃO DE DADOS**

In [18]:
import pandas as pd

df = pd.read_csv('UCMF_raw.csv')
df.head()

ID  Peso  Altura   IMC Atendimento        DN  IDADE Convenio   PULSOS  \
0   1   5.0      51  19.0    11/05/06  30/03/06   0.12       GS  Normais   
1   2   3.5      50  14.0    25/05/05  19/05/05   0.02       GS  Normais   
2   3   0.0       0   NaN    12/06/01  08/05/05  -4.05     SULA  Normais   
3   4   8.1      65  19.0    15/10/09  21/04/09   0.50      NaN  Normais   
4   7  40.0     151  18.0    14/01/08  14/08/95  12.89     SAME  Normais   

   PA SISTOLICA  ...            PPA NORMAL X ANORMAL      B2      SOPRO   FC  \
0           NaN  ...  Não Calculado          Anormal  Normal  Sistólico  112   
1           NaN  ...  Não Calculado          Anormal  Normal    ausente  128   
2           NaN  ...  Não Calculado          Anormal  Normal  Sistólico   88   
3           NaN  ...  Não Calculado          Anormal  Normal    ausente   92   
4           NaN  ...  Não Calculado          Anormal  Normal    ausente   96   

            HDA 1 HDA2 SEXO                      MOTIVO1  \
0      Palpitacao  NaN    M  6 - Suspeita de cardiopatia   
1        Dispneia  NaN    M  6 - Suspeita de cardiopatia   
2   Assintomático  NaN    M                 2 - Check-up   
3   Assintomático  NaN    M     5 - Parecer cardiológico   
4  Dor precordial  NaN    M     5 - Parecer cardiológico   

                               MOTIVO2  
0  6 - Palpitação/taquicardia/arritmia  
1                         6 - Dispnéia  
2                                  NaN  
3                                  NaN  
4                                  NaN  

[5 rows x 21 columns]

Tamanho do conjunto de dados é:

In [66]:
len(df)

12873

## Retirando a coluna *ID*

In [26]:
df_copy = df.copy()
df_copy.drop(columns=['ID'], inplace=True)
df_copy.head()

Peso  Altura   IMC Atendimento        DN  IDADE Convenio   PULSOS  \
0   5.0      51  19.0    11/05/06  30/03/06   0.12       GS  Normais   
1   3.5      50  14.0    25/05/05  19/05/05   0.02       GS  Normais   
2   0.0       0   NaN    12/06/01  08/05/05  -4.05     SULA  Normais   
3   8.1      65  19.0    15/10/09  21/04/09   0.50      NaN  Normais   
4  40.0     151  18.0    14/01/08  14/08/95  12.89     SAME  Normais   

   PA SISTOLICA  PA DIASTOLICA            PPA NORMAL X ANORMAL      B2  \
0           NaN            NaN  Não Calculado          Anormal  Normal   
1           NaN            NaN  Não Calculado          Anormal  Normal   
2           NaN            NaN  Não Calculado          Anormal  Normal   
3           NaN            NaN  Não Calculado          Anormal  Normal   
4           NaN            NaN  Não Calculado          Anormal  Normal   

       SOPRO   FC           HDA 1 HDA2 SEXO                      MOTIVO1  \
0  Sistólico  112      Palpitacao  NaN    M  6 - Suspeita de cardiopatia   
1    ausente  128        Dispneia  NaN    M  6 - Suspeita de cardiopatia   
2  Sistólico   88   Assintomático  NaN    M                 2 - Check-up   
3    ausente   92   Assintomático  NaN    M     5 - Parecer cardiológico   
4    ausente   96  Dor precordial  NaN    M     5 - Parecer cardiológico   

                               MOTIVO2  
0  6 - Palpitação/taquicardia/arritmia  
1                         6 - Dispnéia  
2                                  NaN  
3                                  NaN  
4                                  NaN

## Análise de dados faltantes nas colunas

### Peso

In [44]:
print("Informações gerais sobre a coluna 'Peso':")
print(df_copy['Peso'].describe())

print("\n"+"-"*20)

print("\nNúmero de valores faltantes na coluna 'Peso':")
print(df_copy['Peso'].isna().sum())

Informações gerais sobre a coluna 'Peso':
count    12555.000000
mean        16.614417
std         16.979814
min        -40.000000
25%          3.400000
50%         12.800000
75%         24.000000
max        157.000000
Name: Peso, dtype: float64

--------------------

Número de valores faltantes na coluna 'Peso':
318


Temos por volta de 2.5% dos dados dessa coluna faltante, não é muito preocupante. Preocupante é peso negativo e outros dados estranhos.

### Altura

In [37]:
print("Informações gerais sobre a coluna 'Altura':")
print(df_copy['Altura'].describe())

print("\n"+"-"*20)

print("\nNúmero de valores faltantes na coluna 'Altura':")
print(df_copy['Altura'].isna().sum())

Informações gerais sobre a coluna 'Altura':
count    12873.000000
mean        66.111474
std         56.027934
min          0.000000
25%          0.000000
50%         69.000000
75%        112.000000
max        198.000000
Name: Altura, dtype: float64

--------------------

Número de valores faltantes na coluna 'Altura':
0


Não temos dados faltantes, isso é muito bom. Nem alturas negativas e dados estranhos nessa primeira análise.

### IMC

In [46]:
print("Informações gerais sobre a coluna 'IMC':")
print(df_copy['IMC'].describe())

print("\n"+"-"*20)

print("\nNúmero de valores faltantes na coluna 'IMC':")
print(df_copy['IMC'].isna().sum())

Informações gerais sobre a coluna 'IMC':
count    8146.000000
mean       17.455315
std        15.008016
min         0.000000
25%        15.000000
50%        17.000000
75%        19.000000
max       848.000000
Name: IMC, dtype: float64

--------------------

Número de valores faltantes na coluna 'IMC':
4727


Nessa coluna, possuímos 36.72% dos dados faltantes, isso já começa a ficar preocupante.

### Atendimento

In [49]:
print("\nNúmero de valores faltantes na coluna 'Atendimento':")
print(df_copy['Atendimento'].isna().sum())


Número de valores faltantes na coluna 'Atendimento':
983


Nessa coluna, temos cerca de 7.6% de dados faltantes. Não vejo isso como problema, dado que são poucos dados faltantes e não vejo muita importância para essa coluna (data da visita do paciente).

### DN (Data de Nascimento)

In [52]:
print("\nNúmero de valores faltantes na coluna 'DN':")
print(df_copy['DN'].isna().sum())


Número de valores faltantes na coluna 'DN':
1376


Nessa coluna, temos cerca de 10.7% de dados faltantes. Não vejo isso como problema, dado que são poucos dados faltantes e não vejo muita importância para essa coluna, já que possuímos uma coluna de idade (um pouco suspeita pela qualidade dos dados) e não sabemos quando esse conjunto de dados foi coletado, para fazer o cálculo das idades.

### Idade

In [55]:
print("Informações gerais sobre a coluna 'IDADE':")
print(df_copy['IDADE'].describe())

print("\n"+"-"*20)

print("\nNúmero de valores faltantes na coluna 'IDADE':")
print(df_copy['IDADE'].isna().sum())

Informações gerais sobre a coluna 'IDADE':
count    11377.000000
mean         4.856947
std          8.829742
min       -113.180000
25%          0.930000
50%          3.960000
75%          8.590000
max         71.810000
Name: IDADE, dtype: float64

--------------------

Número de valores faltantes na coluna 'IDADE':
1496


Temos cerca 11.5% de dados faltantes. Porém esse não é o maior problema, esses dados estão muito confusos e as informações que tenho sobre o conjunto de dados não colaboram para entendê-los. Temos uma idade de -113.18, de 0.01, etc. Como posso interpretar isso?

### Convênio

In [61]:
print("\nNúmero de valores faltantes na coluna 'Convenio':")
print(df_copy['Convenio'].isna().sum())


Número de valores faltantes na coluna 'Convenio':
4108


Temos cerca de 32% de dados faltantes nessa coluna. No entanto, isso não é preocupantes, pois ainda não vejo importância dessa coluna para o nosso objetivo, provavelmente vou retirá-la.

### Pulsos

In [60]:
print("\nNúmero de valores faltantes na coluna 'PULSOS':")
print(df_copy['PULSOS'].isna().sum())


Número de valores faltantes na coluna 'PULSOS':
1194


Temos cerca de 9% de dados faltantes.

### PA Sistólica e Diastólica

In [64]:
print("\nNúmero de valores faltantes na coluna 'PA SISTOLICA':")
print(df_copy['PA SISTOLICA'].isna().sum())
print(7730/len(df_copy) * 100)

print("\n"+"-"*20)

print("\nNúmero de valores faltantes na coluna 'PA DIASTOLICA':")
print(df_copy['PA DIASTOLICA'].isna().sum())
print(7730/len(df_copy) * 100)


Número de valores faltantes na coluna 'PA SISTOLICA':
7730
60.04816282140914

--------------------

Número de valores faltantes na coluna 'PA DIASTOLICA':
7740
60.04816282140914


Temos cerca de 60% de dados faltantes 